CB-231: Support for testing MusicBrainz-dependent code

ferbncode opened the pull request:
Added tests for `place` entity and `sample_data.py` which contains all the test data. Presently, the tests take quite sometime to run (working on it).

alastair commented on docker/docker-compose.dev.yml:
```diff
@@ -49,3 +50,12 @@ services:
     ports:
       - "5430:5432"
+
+  musicbrainz_test_db:
```
Is it possible here to use https://hub.docker.com/r/metabrainz/musicbrainz-test-database/ ? (users already have it downloaded so that they can run the musicbrainz database)

ferbncode commented:
I've updated the dockerfiles for using the `musicbrainz-test-database` docker image. That image takes care of making the database as well as dumping the `initial.sql` (https://github.com/metabrainz/musicbrainz-server/blob/master/t/sql/initial.sql). For the test data, I've added the `create_test_data.sql`.
Also, `conftest.py` has been added for populating and deleting the `musicbrainz_test` db at the start and end of a test session.

ferbncode commented:
The following entities are present in the test data:

**Release Groups:** (and their corresponding releases, media and recordings)

- [Numb/Encore](https://critiquebrainz.org/release-group/d5a2e439-85a0-3058-ab9f-ce3be793c805) (has multiple artists)
- [Hybrid Theory EP](https://critiquebrainz.org/release-group/ed3e5734-7a24-32ae-9da0-c6c4b3057109) (so that we have two different types of release groups for the same artist)

**Artists**:

- [Linkin Park](https://critiquebrainz.org/artist/f59c5520-5f46-4d2c-b2c4-822eabf53419?release_type=album) (have artist-rels, url-rels etc)
- [The Spyro](https://critiquebrainz.org/aritst/59b55c37-cb82-4c18-a805-49f486d8a1ae) (Performer at event in the test db)
- Related artists (members of LP, Jay Z)

**Place**

- [Suisto](https://www.critiquebrainz.org/place/d71ffe38-5eaf-426b-9a2e-e1f21bc84609) (has url-rels, place-rels (has backward relation with [Verkatehdas](https://www.critiquebrainz.org/place/f9587914-8505-4bd1-833b-16a3100a4948)))
- [Verkatehdas](https://www.critiquebrainz.org/place/f9587914-8505-4bd1-833b-16a3100a4948)

**Event**

- [The Spyro at Suisto](https://www.critiquebrainz.org/event/f3dc0b63-5d33-4acf-a4fc-d36f45ab1b4f) (Event at Suisto)

gentlecat commented on critiquebrainz/frontend/external/musicbrainz_db/testing.py:
```diff
@@ -0,0 +1,12 @@
+from flask_testing import TestCase
+from critiquebrainz.frontend import create_app
+
+class MBDatabaseTestCase(TestCase):
```
Does this base class need to be based on `flask_testing`'s TestCase or a regular one from `unittest` would be enough?

ferbncode replied:
Yes, `flask_testing`'s TestCase is required as it sets up the `critiquebrainz_redis` container for the tests. I get the following error when using `unittest.TestCase` > https://gist.github.com/ferbncode/2a3f3967015747c45efedfc20b7f3608

gentlecat replied:
If creating a whole Flask app is unnecessary then I'd say it's better to just initialise the cache module.

gentlecat commented on critiquebrainz/frontend/external/musicbrainz_db/tests/helpers_test.py:
```diff
@@ -0,0 +1,48 @@
+import critiquebrainz.frontend.external.musicbrainz_db as mb
+from critiquebrainz.frontend.external.musicbrainz_db.test_data import linkplaceurl_1, linkplaceurl_2, place_suisto
+
+class HelpersTestClass(MBDatabaseTestCase):
```
HelpersTest**Case**

gentlecat commented on critiquebrainz/frontend/external/musicbrainz_db/tests/__init__.py:
```diff
@@ -0,0 +1,14 @@
+
+REDIS_HOST = 'critiquebrainz_redis'
+REDIS_PORT = 6379
+REDIS_NAMESPACE = 'CB'
```
Hmm, this is not ideal. It would be good to get these values from the test configuration instead.
